internal/framework5provider/number_function_test.go

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package framework
+
+import (
+	"fmt"
+	"math/big"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-framework/providerserver"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov5"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/knownvalue"
+	"github.com/hashicorp/terraform-plugin-testing/plancheck"
+	"github.com/hashicorp/terraform-plugin-testing/statecheck"
+	"github.com/hashicorp/terraform-plugin-testing/tfversion"
+)
+
+func TestNumberFunction_known(t *testing.T) {
+	// Using 9223372036854775808, the smallest number that can't be represented as an int64,
+	// results in an Terraform error where [Planned value does not match config value for cty.NumberIntVal],
+	// which is related to a bug in go.cty relating to [Large integer comparisons and msgpack encoding].
+	// A value of 9223372036854775809 is used for the meanwhile.
+	//
+	// [Planned value does not match config value for cty.NumberIntVal]: https://github.com/hashicorp/terraform/issues/34589
+	// [Large integer comparisons and msgpack encoding]: https://github.com/zclconf/go-cty/pull/176
```

Comment:
I super love/appreciate this additional code comment context! Looks like this has been closed out upstream and should be resolved as of Terraform 1.8.0-beta1, which I think means we can remove this comment now, but curious if you think this context needs to hang out anywhere?

Reply:
Agree on the removal. I've also reverted the integer that's used in the test to `9223372036854775808`.

I don't think we need to explicitly add anything relating to this issue/discovery within the terraform-provider-corner repository, and there's an [issue](https://github.com/hashicorp/terraform-plugin-framework/issues/943) which captures the context in terraform-plugin-framework, which I believe should be sufficient.